### PR TITLE
feat(stark-all): integrate SonarTS to current TSLint checks. Refactor code to fix TSLint violations.

### PR DESCRIPTION
**Pull request by christophercr**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[X] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Current TSLint rules focus mostly on formatting and common code styling issues. This however does not provide enough checks to detect bugs and/or suspicious patterns.

Issue Number: #331

## What is the new behavior?
Stricter TSLint rules focused on bug detection and code smell detection in order to ensure better code quallity.

For now, these rules have been disabled until the affected code is refactored:

- [cognitive-complexity](https://github.com/SonarSource/SonarTS/blob/master/sonarts-core/docs/rules/cognitive-complexity.md)
- [no-duplicate-string](https://github.com/SonarSource/SonarTS/blob/master/sonarts-core/docs/rules/no-duplicate-string.md)
- [no-big-function](https://github.com/SonarSource/SonarTS/blob/master/sonarts-core/docs/rules/no-big-function.md)
- [no-commented-code](https://github.com/SonarSource/SonarTS/blob/master/sonarts-core/docs/rules/no-commented-code.md)

**IMPORTANT:** By the moment, the presets we use were moved to the tslint.json file in the stark-core package and the starter due to this issue: palantir/tslint#3874

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: packages/stark-build/config/tslint.json [hunk 1]
```diff
@@ -1,7 +1,13 @@
 {
-  "extends": ["tslint:latest", "tslint-config-prettier", "codelyzer"],
+  // FIXME: presets moved to packages/stark-core/tslint.json temporarily due to this issue: https://github.com/palantir/tslint/issues/3874
   "defaultSeverity": "error",
   "rules": {
+    // sonarts rules
```
Comment (dsebastien) [inline packages/stark-build/config/tslint.json]: Can you create an issue to track these and enable them later on?

File: packages/stark-build/config/webpack.dev.js [hunk 2]
```diff
@@ -36,296 +36,298 @@ module.exports = function (env) {
...
 		 *
 		 * See: https://webpack.github.io/docs/webpack-dev-server.html
 		 */
-		before: function (app) {
-			// For example, to define custom handlers for some paths:
-			// app.get('/some/path', function(req, res) {
-			//   res.json({ custom: 'response' });
-			// });
-		},
+		devServer: {
```
Comment (dsebastien) [inline packages/stark-build/config/webpack.dev.js]: All the styling changes are really annoying in our PRs. I think we should isolate those in separate PRs if possible.

**Comment by dsebastien** [comment]
Nice improvement, this’ll help us a lot soon :)

File: packages/stark-build/config/tslint.json (hunk 1 above)
Comment (christophercr) [reply packages/stark-build/config/tslint.json]: I've created this issue: #360

**Comment by christophercr** [comment]
Created issue #360 to track the enabling of the stricter SonarTS rules

**Pull request merged**